idaplugin/rematch/dialogs/matchresult.py

```diff
@@ -0,0 +1,378 @@
+    self.btn_filter.clicked.connect(self.filter)
+    self.btn_apply.clicked.connect(self.apply_matches)
+
+    # matches tree
+    self.search_box = QtWidgets.QLineEdit()
+    self.tree = SearchTreeWidget(search_box=self.search_box,
+                                 match_column=self.MATCH_NAME_COLUMN)
+
+    # tree columns
+    self.tree.setHeaderLabels(("Function", "Score", "Doc. Score", "Engine"))
+
+    self.tree.header().setDefaultSectionSize(20)
+    self.tree.resizeColumnToContents(self.MATCH_SCORE_COLUMN)
+    self.tree.resizeColumnToContents(self.DOCUMENTATION_SCORE_COLUMN)
+    self.tree.setColumnWidth(self.MATCH_NAME_COLUMN, 150)
+
+    # other tree properties
+    self.tree.setFrameShape(QtWidgets.QFrame.NoFrame)
+    self.tree.setAlternatingRowColors(True)
+    self.tree.setSortingEnabled(True)
+    self.tree.sortItems(self.MATCH_SCORE_COLUMN, QtCore.Qt.DescendingOrder)
+
+    # text browser (code highlighting & display)
+    self.textBrowser = QtWidgets.QTextBrowser()
+    self.textBrowser.zoomIn(2)
+
+    # splitter for code highlighting and tree
+    self.frame = QtWidgets.QFrame()
+    self.frame.sizeHint = lambda: QtCore.QSize(-1, -1)
+    self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
+    self.vlayoutMainFrame = QtWidgets.QVBoxLayout()
+    self.vlayoutMainFrame.addWidget(self.tree)
+    self.vlayoutMainFrame.addWidget(self.search_box)
+    self.vlayoutMainFrame.addLayout(self.hlayoutButtons)
+    self.frame.setLayout(self.vlayoutMainFrame)
+
+    self.splitter = QtWidgets.QSplitter()
+    self.splitter.setOrientation(QtCore.Qt.Horizontal)
+    self.splitter.addWidget(self.frame)
+    self.splitter.addWidget(self.textBrowser)
+    self.splitter.setSizes([150, 650])
+
+    # main layout
+    self.base_layout.addWidget(self.splitter)
+
+    # connect events to handle
+    self.tree.itemChanged.connect(self.itemChanged)
+    self.tree.itemSelectionChanged.connect(self.itemSelectionChanged)
+    self.tree.itemDoubleClicked.connect(self.itemDoubleClicked)
+
+    self.populate_tree()
+    self.set_checks()
+
+  def get_obj(self, obj_id):
+    if obj_id in self.locals:
+      return self.locals[obj_id]
+    else:
+      return self.remotes[obj_id]
+
+  def itemSelectionChanged(self):
+    if not self.tree.selectedItems():
+      return
+
+    item = self.tree.selectedItems()[0]
+    parent = item.parent()
+    if parent is None:
+      return
+
+    id1 = item.parent().api_id
+    id2 = item.api_id
+
+    try:
+      response = network.query("GET", "display/compare/",
+                               params={"id1": id1, "id2": id2}, json=False)
+      self.textBrowser.setHtml(response)
+      self.textBrowser.reload()
+    except exceptions.QueryException:
+      pass
+
+  def itemDoubleClicked(self, item, column):
+    del column
+
+    if item.parent() is None:
+      idaapi.jumpto(self.get_obj(item.api_id)['offset'])
+      item.setExpanded(not item.isExpanded())
+
+  def itemChanged(self, item, column):
```

Comment:
fix replace checked makes everything unchecked due to recursive `itemChanged` events